feature-libs/my-account/organization/src/core/guards/admin.guard.ts

```diff
@@ -0,0 +1,24 @@
+import { Injectable } from '@angular/core';
+import { CanActivate } from '@angular/router';
+import { UserService } from '@spartacus/core';
+import { Observable } from 'rxjs';
+import { map, pluck } from 'rxjs/operators';
+import { UnitRoleType } from '../../components/shared';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AdminGuard implements CanActivate {
+  constructor(protected userService: UserService) {}
+
+  canActivate(): Observable<boolean> {
+    return this.userService.get().pipe(
+      pluck('roles'),
+      map((roles: string[]) =>
+        roles && roles.length
+          ? !!roles.find((r) => r === UnitRoleType.ADMIN)
+          : false
```

Comment:
```suggestion
        roles?.length && roles.includes(UnitRoleType.ADMIN)           
```
// or 
`        Array.isArray(roles) && roles.includes(UnitRoleType.ADMIN)`